tests/functional/test_contribute.py

```diff
@@ -0,0 +1,51 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+from selenium.common.exceptions import TimeoutException
+
+from ..pages.contribute import ContributePage
+
+
+@pytest.mark.nondestructive
+def test_open_close_modal(base_url, selenium):
+    page = ContributePage(base_url, selenium).open()
+    page.open_video_modal()
+    assert page.video_modal_is_displayed
+    page.close_video_modal()
+
+
+@pytest.mark.nondestructive
+def test_next_event_is_displayed(base_url, selenium):
+    page = ContributePage(base_url, selenium).open()
+    assert page.next_event_is_displayed
+
+
+@pytest.mark.nondestructive
+def test_default_values(base_url, selenium):
+    page = ContributePage(base_url, selenium).open()
+    page.newsletter.expand_form()
+    assert '' == page.newsletter.email
+    assert 'United States' == page.newsletter.country
+    assert page.newsletter.html_format_selected
+    assert not page.newsletter.text_format_selected
+    assert not page.newsletter.privacy_policy_accepted
+
+
+def test_successful_sign_up(base_url, selenium):
+    page = ContributePage(base_url, selenium).open()
+    page.newsletter.expand_form()
+    page.newsletter.type_email('noreply@mozilla.com')
+    page.newsletter.select_country('United Kingdom')
+    page.newsletter.select_text_format()
+    page.newsletter.accept_privacy_policy()
+    page.newsletter.click_sign_me_up()
+    assert page.newsletter.sign_up_successful
+
+
+@pytest.mark.nondestructive
+def test_sign_up_fails_when_missing_required_fields(base_url, selenium):
+    page = ContributePage(base_url, selenium).open()
+    with pytest.raises(TimeoutException):
+        page.newsletter.click_sign_me_up()
```

Comment:
You might want to tweak the test names, as the last three are specific to the newsletter form on the contribute page but they don't make that clear. Perhaps `test_newsletter_default_values`, `test_successful_newsletter_sign_up` and `test_newsletter_sign_up_fails_when_missing_required_fields`.

Alternatively, you could create a `TestNewsletter` class and move these tests beneath. This would then appear in reports as `test_contribute.py::TestNewsletter.test_default_values`, etc.